packages/js/src/utils/generators.test.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  CODE_VERIFIER_MAX_LENGTH,
+  generateCodeChallenge,
+  generateCodeVerifier,
+  generateState,
+} from './generators';
+
+describe('generateState', () => {
+  test('with fixed length', () => {
+    const state = generateState();
+    expect(state.length).toEqual(CODE_VERIFIER_MAX_LENGTH);
+  });
+
+  test('with random value', () => {
+    const state1 = generateState();
+    const state2 = generateState();
+    expect(state1).not.toEqual(state2);
+  });
+});
+
+describe('generateCodeVerifier', () => {
+  test('with fixed length', () => {
+    const verifier = generateCodeVerifier();
+    expect(verifier.length).toEqual(CODE_VERIFIER_MAX_LENGTH);
+  });
+
+  test('with random value', () => {
+    const verifier1 = generateCodeVerifier();
+    const verifier2 = generateCodeVerifier();
+    expect(verifier1).not.toEqual(verifier2);
+  });
+});
+
+describe('generateCodeChallenge', () => {
```

Comment:
you also need some static [test cases](https://github.com/logto-io/swift/blob/master/Tests/LogtoTests/LogtoUtilitiesTests.swift#L18-L38) to prove the algorithm is correct, actually that's the core

Reply:
Added.

Reply:
i didn't mean 'dealing with different code verifiers should not be equal' and 'dealing with same code verifier should be equal' should use static strings.. 😂 should add a new `test()` to test static values, please see the link above

Reply:
Fixed, PTAL.